widgetssdk/src/main/java/com/glia/widgets/InitializationProvider.kt

```diff
@@ -0,0 +1,43 @@
+package com.glia.widgets
+
+import android.app.Application
+import android.content.ContentProvider
+import android.content.ContentValues
+import android.database.Cursor
+import android.net.Uri
+
+/**
+ * @hide
+ */
+open class InitializationProvider : ContentProvider() {
+    override fun onCreate(): Boolean {
+        (context as? Application)?.let { GliaWidgets.onAppCreate(it) }
+        return true
+    }
+
+    override fun query(
+        p0: Uri,
+        p1: Array<out String>?,
+        p2: String?,
+        p3: Array<out String>?,
+        p4: String?
+    ): Cursor? {
+        return null
+    }
+
+    override fun getType(p0: Uri): String? {
+        return null
+    }
+
+    override fun insert(p0: Uri, p1: ContentValues?): Uri? {
+        return null
+    }
+
+    override fun delete(p0: Uri, p1: String?, p2: Array<out String>?): Int {
+        return 0
+    }
+
+    override fun update(p0: Uri, p1: ContentValues?, p2: String?, p3: Array<out String>?): Int {
+        return 0
+    }
+}
```

Comment:
Not critical, but it might still be better to have descriptive param names instead of `p0`, `p1`, etc.